### PR TITLE
Drop support for non RBAC clusters

### LINKED ISSUE #496: Deploying Ambassador to Kubernetes from getting started guide fails
**Describe the bug**
I was trying out deploying Ambassador to Kubernetes using https://www.getambassador.io/user-guide/getting-started, but my pods fail to come up.

**To Reproduce**
- `kubectl apply -f https://getambassador.io/yaml/ambassador/ambassador-no-rbac.yaml`
-
```console
$ kubectl get pods
NAME                          READY     STATUS             RESTARTS   AGE
ambassador-7d95446b76-b5z2t   1/2       CrashLoopBackOff   10         31m
ambassador-7d95446b76-g6dt7   1/2       CrashLoopBackOff   10         31m
ambassador-7d95446b76-pljv4   1/2       CrashLoopBackOff   10         31m
```
-
```console
$ kubectl logs ambassador-7d95446b76-b5z2t -c ambassador
Traceback (most recent call last):
  File "/application/kubewatch.py", line 493, in <module>
    main()
  File "/usr/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/application/kubewatch.py", line 476, in main
    sync(restarter)
  File "/application/kubewatch.py", line 313, in sync
    for x in v1.list_namespaced_config_map(restarter.namespace).items ]
  File "/usr/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 12395, in list_namespaced_config_map
    (data) = self.list_namespaced_config_map_with_http_info(namespace, **kwargs)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 12497, in list_namespaced_config_map_with_http_info
    collection_formats=collection_formats)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 335, in call_api
    _preload_content, _request_timeout)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 148, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 371, in request
    headers=headers)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/rest.py", line 250, in GET
    query_params=query_params)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/rest.py", line 240, in request
    raise ApiException(http_resp=r)
kubernetes.client.rest.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Date': 'Thu, 07 Jun 2018 13:07:43 GMT', 'Content-Length': '269'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"configmaps is forbidden: User \"system:serviceaccount:default:default\" cannot list configmaps in the namespace \"default\"","reason":"Forbidden","details":{"kind":"configmaps"},"code":403}


AMBASSADOR: kubewatch sync exited with status 1
Here's the envoy.json we were trying to run with:
ls: /etc/envoy*.json: No such file or directory
No config generated.
AMBASSADOR: shutting down
```

**Expected behavior**
Well, it should work :)

**Versions (please complete the following information):**
 - Ambassador: [e.g. 0.32.1]
`datawire/ambassador:0.34.1`
 - Kubernetes environment [e.g. Minikube, bare metal, Google Kubernetes Engine]
```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:44:10Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

$ minikube version
minikube version: v0.27.0
```

### PR DESCRIPTION
**Pull request by containscafeine**
This commit drops support for non-RBAC enabled Kubernetes clusters
since RBAC has been GA since Kubernetes 1.8, and RBAC is now
enabled by default in minikube since release 0.26.0.

Docs have been updated and relevant files have been removed or
renamed.

Fix #496

### CODE REVIEW

File: docs/user-guide/getting-started.md [hunk 1]
```diff
@@ -6,16 +6,10 @@ Ambassador is designed to allow service authors to control how their service is
...
-```
-
-Without RBAC, you can use:
-
-```shell
-kubectl apply -f https://getambassador.io/yaml/ambassador/ambassador-no-rbac.yaml
+kubectl apply -f https://getambassador.io/yaml/ambassador/ambassador.yaml
 ```
 
 For production configurations, we recommend you download these YAML files as your starting point, and customize them accordingly (e.g., your namespace).
```
Comment (richarddli) [inline docs/user-guide/getting-started.md]: Should be singular now, not plural

**Comment by richarddli** [comment]
I wonder if it would be better to be explicit about non-RBAC and still keep it around.

"If you're on an older cluster (e.g., Kubernetes < 1.8 or Minikube < 0.xxx), you might not have RBAC enabled, in which case you should do this."

??? Or do we think 99.99% of people will have RBAC at this point?

**Comment by containscafeine** [comment]
> ??? Or do we think 99.99% of people will have RBAC at this point?

ping @plombardi89

**Comment by plombardi89** [comment]
> ??? Or do we think 99.99% of people will have RBAC at this point?

I doubt the number is that high, but I would be surprised if it's under 75% given this feature went stable last fall and was in use before then with 1.7 clusters. With the way Kubernetes releases work 1.8 is the oldest maintained version of Kubernetes still out there. When 1.11 ships end of June 1.8 will stop receiving updates and most people will be upgrading to something recent (likely 1.10 as it seems few people jump right to the latest release). 

@richarddli do we have any metrics around which manifests are most often used?

**Comment by plombardi89** [comment]
My advice is to err on the side of caution here without good metrics and wait. When Kubernetes 1.12 ships September/October that will mark 1 year since RBAC when default and at that point it would likely be safe to remove or make a footnote.

**Comment by richarddli** [comment]
OK, let's be cautious, and not remove the reference and reassess in 2-3 months. I also searched the Minikube release notes and they don't say anything about RBAC either (?!!!).

**Pull request closed without merging**